### PR TITLE
fix(timeout-type-memo): update nix dependency proper way

### PR DESCRIPTION
**Pull request by RustNinja**
Required for merge:
- [x] `pr-workflow-check / draft-release-check` is ✅ success
- Other rules GitHub shows you, or can be read in [configuration](../terraform/github.com/branches.tf) 

Makes review faster:
- [x] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [x] I marked PR by `misc` label if it should not be in release notes
- [x] Linked Zenhub/Github/Slack/etc reference if one exists
- [x] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [x] Added reviewer into `Reviewers`
- [x] I tagged(`@`) or used other form of notification of one person who I think can handle best review of this PR
- [ ] I have proved that PR has no general regressions of relevant features and processes required to release into production
- [ ] Any dependency updates made, was done according guides from relevant dependency
- Clicking all checkboxes 
- Adding detailed description of changes when it feels appropriate (for example when PR is big)

### CODE REVIEW

**Comment by kkast** [comment]
test locally

**Review by kkast — CHANGES_REQUESTED** [review]
waiting for local testing

File: code/parachain/frame/composable-traits/src/centauri.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ impl Map {
 				receiver: value.receiver,
 				port: value.port.map(|x| x.to_string()),
 				channel: value.channel.map(|x| x.to_string()),
-				timeout: value.timeout,
+				timeout: value.timeout.and_then(|x| x.parse().ok()),
```
Comment (vmarkushin) [inline code/parachain/frame/composable-traits/src/centauri.rs:16]: don't we want to fail the whole memo if this fails?

Comment (RustNinja) [reply code/parachain/frame/composable-traits/src/centauri.rs:16]: we know 100% that the value is number
```
ForwardingMemo::new_ibc_memo(
      memo_receiver,
      PortId::transfer(),
      ChannelId::new(i.channel_id),
      i.timeout.ok_or(DispatchError::Other("Timeout is none"))?.to_string(),
      i.retries.ok_or(DispatchError::Other("Retries is none"))?,
)
```

Comment (vmarkushin) [reply code/parachain/frame/composable-traits/src/centauri.rs:16]: But what if the realisation changes and someone accidentally put not a number there?

Comment (RustNinja) [reply code/parachain/frame/composable-traits/src/centauri.rs]: pushed the changes. to replace string with u64. thanks.

**Review by vmarkushin — APPROVED** [review]

**Review by kkast — CHANGES_REQUESTED** [review]

File: code/parachain/frame/pallet-multihop-xcm-ibc/src/lib.rs [hunk 2]
```diff
@@ -303,7 +303,7 @@ pub mod pallet {
 		}
 	}
 
-	impl<T: Config> Pallet<T> {
+	impl<T> Pallet<T> {
```
[the remark this answers is not in the document]
Comment (RustNinja) [reply code/parachain/frame/pallet-multihop-xcm-ibc/src/lib.rs:306]: we do not need this trait bound.

**Comment by dzmitry-lahoda** [comment]
@RustNinja may you elaborate on on how check was concluded that regression did not happen?
If if was not tested, what is step by step guide to do so?

image

**Review by kkast — APPROVED** [review]
tested

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**